syntax = "proto3";
package collection_config;




import "google/protobuf/struct.proto";

/*
CreateCollectionConfig请求
*/
message CreateCollectionConfigRequest {
message Kwargs {
    /*
    键
    */
    string key = 1 ;
    /*
    值
    */
    google.protobuf.Value value = 2 ;
}
        
message Env {
    /*
    键
    */
    string key = 1 ;
    /*
    值
    */
    google.protobuf.Value value = 2 ;
}
        
message Script {
    /*
    脚本Id
    */
    string id = 1 ;
    /*
    脚本版本
    */
    string version = 2 ;
    /*
    脚本类型
    */
    string type = 3 ;
    /*
    脚本内容
    */
    string content = 4 ;
    /*
    脚本名称
    */
    string name = 5 ;
}
        
message HostRange {
message Query {
    /*
    键
    */
    string key = 1 ;
    /*
    操作符
    */
    string op = 2 ;
    /*
    值
    */
    google.protobuf.Value value = 3 ;
}
        
    
    /*
    模型ID
    */
    string objectId = 1 ;
    /*
    获取主机Key路径
    */
    string hostKey = 2 ;
    /*
    查询者
    */
    string querier = 3 ;
    /*
    查询类型
    */
    string type = 4 ;
    /*
    查询策略字符串
    */
    string queryString = 5 ;
    /*
    查询策略ID
    */
    string queryId = 6 ;
    /*
    查询策略ID
    */
    repeated Query query = 7 ;
}
        
    
    /*
    名称
    */
    string name = 1 ;
    /*
    参数
    */
    repeated Kwargs kwargs = 2 ;
    /*
    超时时间
    */
    int32 timeout = 3 ;
    /*
    环境变量
    */
    repeated Env env = 4 ;
    /*
    是否禁用
    */
    bool disabled = 5 ;
    /*
    标签
    */
    repeated string labels = 6 ;
    /*
    脚本
    */
    Script script = 7 ;
    /*
    是否忽略不符合要求数据
    */
    bool ignoreInvalid = 8 ;
    /*
    主机范围
    */
    HostRange hostRange = 9 ;
    /*
    采集间隔
    */
    int32 interval = 10 ;
    /*
    缓存过期次数（时间）
    */
    int32 cacheTtl = 11 ;
    /*
    执行时间范围（24小时）
    */
    string timeRange = 12 ;
    /*
    采集类型ID
    */
    string clazzId = 13 ;
    /*
    关联模型ID
    */
    string objectId = 14 ;
    /*
    关联模型实例ID宏
    */
    string instanceIdMacro = 15 ;
}

/*
CreateCollectionConfig返回
*/
message CreateCollectionConfigResponse {
    /*
    ID
    */
    string id = 1 ;
}

/*
CreateCollectionConfigApi返回
*/
message CreateCollectionConfigResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    collection_config.CreateCollectionConfigResponse data = 4 ;
}

